### PR TITLE
[JENKINS-47178] Replace Windows path separator with Unix path separator

### PR DESCRIPTION
**Pull request by TheOriginalAlex**
Use Unix path separator in volume mount paths

Signed-off-by: Alex Johnson <ajohnson@bombora.com>

### CODE REVIEW

**Comment by Vlatombe** [comment]
What issue is this fixing? Would be great to add a test as well.

**Comment by TheOriginalAlex** [comment]
@Vlatombe: If the Jenkins master is running on a Windows machine, the `Path.toString` function uses the Windows path separator in volume mount paths.

**Comment by TheOriginalAlex** [comment]
```
volumeMounts:
    - mountPath: \usr\bin\docker
      name: volume-2
    - mountPath: \root\workspace
      name: volume-0
    - mountPath: \var\run\docker.sock
      name: volume-1
```

**Comment by TheOriginalAlex** [comment]
@Vlatombe: And regarding tests, are there any tests that are currently testing volume creation?

**Comment by Vlatombe** [comment]
@TheOriginalAlex Thanks for providing the context.
If I understand the issue correctly, it means that the existing test class `org.csanchez.jenkins.plugins.kubernetes.PodTemplateBuilderTest` fails on windows and should be fixed by this PR. Could you confirm?

@carlossg I guess the best way to handle such problems would be to have a tests running on windows.

**Comment by carlossg** [comment]
That test is already passing on windows, maybe needs some extra checks
https://ci.jenkins.io/job/Plugins/job/kubernetes-plugin/job/master/173/testReport/org.csanchez.jenkins.plugins.kubernetes/PodTemplateBuilderTest/

**Comment by Vlatombe** [comment]
@carlossg ah right, it's loading definition from yaml files, so it doesn't go through the path getting changed here.

Probably just a matter of creating a new test case that is declaring a volume mount using the `PodTemplateBuilder` methods directly.

**Comment by TheOriginalAlex** [comment]
Wouldn't making this `setYaml` function in `org.csanchez.jenkins.plugins.kubernetes.PodTemplate` load contents of the yaml into the object properties (i.e. `this.volumes`) ensure that the existing tests catch these errors?
```
@DataBoundSetter
    public void setYaml(String yaml) {
        this.yaml = yaml;
    }
```

**Comment by TheOriginalAlex** [comment]
Or is there a specific reason why only the string version of the yaml is stored in the object?

**Comment by Vlatombe** [comment]
The yaml allows to support any kind of field supported by kubernetes that may not be implemented by the plugin.

**Comment by TheOriginalAlex** [comment]
@Vlatombe: Got it!  Thanks for the clarification!

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilderTest.java [hunk 1]
```diff
@@ -1,6 +1,12 @@
 package org.csanchez.jenkins.plugins.kubernetes;
 
 import static org.csanchez.jenkins.plugins.kubernetes.PodTemplateBuilder.*;
+import static org.csanchez.jenkins.plugins.kubernetes.volumes.HostPathVolume.*;
+import static org.csanchez.jenkins.plugins.kubernetes.volumes.EmptyDirVolume.*;
+import static org.csanchez.jenkins.plugins.kubernetes.volumes.PodVolume.*;
+import static org.csanchez.jenkins.plugins.kubernetes.volumes.ContainerTemplate.*;
```
Comment (carlossg) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilderTest.java]: wrong package name

**Comment by carlossg** [comment]
the test is failing with

```
ERROR] testBuildFromTemplate(org.csanchez.jenkins.plugins.kubernetes.PodTemplateBuilderTest)  Time elapsed: 0.005 s  <<< FAILURE!

java.lang.AssertionError

	at org.junit.Assert.fail(Assert.java:86)

	at org.junit.Assert.assertTrue(Assert.java:41)

	at org.junit.Assert.assertNotNull(Assert.java:712)

	at org.junit.Assert.assertNotNull(Assert.java:722)

	at org.csanchez.jenkins.plugins.kubernetes.PodTemplateBuilderTest.validatePod(PodTemplateBuilderTest.java:155)

	at org.csanchez.jenkins.plugins.kubernetes.PodTemplateBuilderTest.testBuildFromTemplate(PodTemplateBuilderTest.java:128)

```

**Comment by TheOriginalAlex** [comment]
@carlossg: Finally got around to fixing the tests... sorry about the delay!

**Comment by carlossg** [comment]
thanks!

**Pull request merged**